cli/src/git_util.rs

```diff
@@ -535,3 +544,40 @@ fn warn_if_branches_not_found(
 
     Ok(())
 }
+
+const DEFAULT_REMOTE: &str = "origin";
+pub fn get_fetch_remotes(
+    ui: &Ui,
+    settings: &UserSettings,
+    git_repo: &git2::Repository,
+    remotes: &[String],
+    use_all_remotes: bool,
```

Comment:
If both `jj git sync` and `jj git fetch` share command parameters, it might be better to extract common `Args` struct. Then, we can pass it in to helper functions.

Reply:
Makes sense.

I'm thinking this could actually collapse down to something like:

```rust
let fetcher = GitFetcher::new(ui, tx, git_repo, branch, remotes, use_all_remotes);
fetcher.fetch()?;
```

Inside `fetcher.fetch` we'd grab the remotes and then do the fetch. Mainly because I can't think of a good reason why to expose to function calls to the commands instead of just one, as the intermediate result is not used by callers.

Does that feel like overkill?

In the meantime, I'm thinking to merge this PR and make changes in a follow-up to keep things clearer.

Let me know if you think otherwise.

Reply:
I'm not sure. Do we have more methods than `fetch()`?

My concern about the current API is that it's hard to tell which `String`/`bool` argument represents which variable.

Reply:
> I'm not sure. Do we have more methods than `fetch()`?

I just realized we do need to expose `get_fetch_remotes`, since we use the result to build a message after the fetch succeeds further down.

> 
> My concern about the current API is that it's hard to tell which `String`/`bool` argument represents which variable.

I've added a FetchArgs struct that should make this better and doesn't end up feeling over designed.

wdyt?